.github/workflows/test.yml

```diff
@@ -1,12 +1,10 @@
 name: test
 
 on:
-  push:
```

Comment:
We require code to be re-based on develop before merging so it's not possible for this to fail after merging if the CI passes on the PR. Flakes aside of course, but for broken code our process doesn't allow it.

This is how we had things set up previously when everything was using Nix. I suggested the change to save capacity and that's even more important now that we have a limited number of mac jobs concurrently.